doc/rtd/reference/merging.rst

```diff
@@ -29,88 +19,191 @@ For example:
+These options apply to lists:
 
-- :command:`append`: Add new value to the end of the list. Defaults to
-  ``False``.
-- :command:`prepend`: Add new values to the start of the list. Defaults to
-  ``False``.
+- :command:`append`: Add new value to the end of the list. Disabled by 
+  default.
+- :command:`prepend`: Add new values to the start of the list. Disabled by
+  default.
 - :command:`no_replace`: Do not replace an existing value if one is already
   present. Enabled by default.
-- :command:`replace`: Overwrite existing values with new ones.
+- :command:`replace`: Overwrite existing values with new ones. Disabled by
+  default.
 
-String
-------
+Strings
+~~~~~~~
 
-The :command:`Str` merger has the following options, which control what is
-done with the values contained within the config.
+These options apply to strings:
 
-- :command:`append`: Add new value to the end of the string. Defaults to
-  False.
+- :command:`append`: Add new value to the end of the string. Disabled by 
+  default.
 
 Common options
---------------
+~~~~~~~~~~~~~~
 
 These are the common options for all merge types, which control how recursive
 merging is done on other types.
 
-- :command:`recurse_dict`: If ``True``, merge the new values of the
-  dictionary. Defaults to ``True``.
-- :command:`recurse_list`: If ``True``, merge the new values of the list.
-  Defaults to ``False``.
+- :command:`recurse_dict`: Merge the new values of the dictionary. Enabled by
+  default.
+- :command:`recurse_list`: Merge the new values of the list. Disabled by 
+  default.
 - :command:`recurse_array`: Alias for ``recurse_list``.
-- :command:`recurse_str`: If ``True``, merge the new values of the string.
-  Defaults to False.
+- :command:`recurse_str`: Merge the new values of the string. Disabled by 
+  default.
+
+Example cloud-config
+====================
+
+A common request is to include multiple ``runcmd`` directives in different
+files and merge all of the commands together. To achieve this, we must modify
+the default merging to allow for dictionaries to join list values.
+
+The first config:
+
+.. code-block:: yaml
+
+   #cloud-config
+   merge_how:
+    - name: list
+      settings: [append]
+    - name: dict
+      settings: [no_replace, recurse_list]
+
+   runcmd:
+     - bash1
+     - bash2
+
+The second config:
+
+.. code-block:: yaml
+
+   #cloud-config
+   merge_how:
+    - name: list
+      settings: [append]
+    - name: dict
+      settings: [no_replace, recurse_list]
+
+   runcmd:
+     - bash3
+     - bash4
+
+The effective config:
+
+.. code-block:: yaml
+
+   #cloud-config
+   runcmd:
+     - bash1
+     - bash2
+     - bash3
+     - bash4
+
+Specifying multiple merge types
+===============================
+
+When several cloud-config files define a merge type, the effective type will
+be the result of stacking all previous types, starting from the default:
+
++---+---------------+--------------------------+---------------------------------------------------+
```

Comment:
Sorry, I misunderstood your earlier comment. It's not that the various pieces will individually stack; rather if you haven't specified any merge strategy at all, it'll continue to use the previous one (as a whole).

Another example:
1:
```
#cloud-config
phone_home:
  url: "http://mysite.com/"
```
2:
```
#cloud-config
phone_home:
  url: "some/path/"

merge_how: "dict(recurse_str)+str(append)"
```
After these two, url will be `http://mysite.com/some/path` . If we add a 3rd config
```
#cloud-config
phone_home:
  url: "even/more/path"
merge_how: "dict(recurse_str)"
```
We get `http://mysite.com/some/path`. The `str(append)` from config 2 doesn't apply here.

The extra confusing part (that I'll admit I didn't understand until digging into this now), is that `dict(recurse_str)` is a different state from `dict(recurse_str)+str()`. The default of `list()+dict()+str()` only applies at the top-level. If you start recursing down a structure, a value won't be merged in unless it has a type explicitly specified.

So if we go back to the last config of my last example (only showing the resulting `url` for brevity):
- `merge_how: "dict(recurse_str)"`:  `http://mysite.com/some/path`
- `merge_how: "dict(recurse_str)+str()"`: `even/more/path`
- `merge_how: "dict(recurse_str)+str(append)"`: `http://mysite.com/some/path/even/more/path`

For this section, I think it's fine to simply state that any new config parts will use the most recently specified merge strategy. I can add some extra documentation later trying to clarify the behavior I mentioned here.